src/tactic/core.lean

```diff
@@ -497,17 +497,35 @@ meta def revert_deps (e : expr) : tactic ℕ := do
 meta def is_local_def (e : expr) : tactic unit :=
 retrieve $ do revert e, expr.elet _ _ _ _ ← target, skip
 
+private def partition_local_deps_aux {α} [decidable_eq α] (vs : list α) : list α → list α → list (list α)
+| [] acc := [acc.reverse]
+| (l :: ls) acc :=
+  if l ∈ vs then acc.reverse :: partition_local_deps_aux ls [l]
+  else partition_local_deps_aux ls (l :: acc)
+
+/-- `partition_local_deps vs`, with `vs` a list of local constants,
+reorders `vs` in the order they appear in the local context together
+with the variables that follow them. If local context is `[a,b,c,d,e,f]`,
+and that we call `partition_local_deps [d,b]`, we get `[[b,c],[d,e,f]]`.
+The head of each list is one of the variables given as a parameter. -/
+meta def partition_local_deps (vs : list expr) : tactic (list (list expr)) :=
+do ls ← local_context,
+   pure (partition_local_deps_aux vs ls []).tail.reverse
```

Comment:
Tempting to make `ls` an argument which makes this function non-meta too, although I'm not sure it matters much.

Reply:
I considered it. I don't think the behavior is general enough that I would use it on anything but `local_context`. Another indication that it is very specialized is that in `clear_value`, if I needed the lists in reverse order, I would put that as part of `partition_local_deps`. I think it would fit it nicely as a submodule of the tactic but not as a general operation on lists